### PR TITLE
[INFRA/CORE] Feature add disasterperiod to clientconf

### PR DESCRIPTION
**Pull request by Eirikalb**
# Summary

Add disasterPeriod to client config as a selectively available value.

## Additional Information

As discussed with @JamesTev earlier today
## Test Plan

Added one testcase which seems to pass 👍

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

File: internal/common/config/clientconfig.go [hunk 1]
```diff
@@ -13,6 +13,8 @@ type ClientConfig struct {
 // ClientDisasterConfig contains disaster config information visible to clients.
 type ClientDisasterConfig struct {
 	CommonpoolThreshold SelectivelyVisibleResources
+	DisasterPeriod      SelectivelyVisibleDisasterPeriod
+	StochasticDisasters bool // if true, period between disasters becomes random. If false, it will be consistent (deterministic)
```
Comment (JamesTev) [inline internal/common/config/clientconfig.go]: Why is this not selectively visible too?

**Review by JamesTev — APPROVED** [review]
Looks good, thanks @Eirikalb. Haven't looked through this selective visibility config code in detail but back you

File: internal/common/config/helpertypes.go [hunk 2]
```diff
@@ -25,6 +25,16 @@ type SelectivelyVisibleResources struct {
 	Valid bool
 }
 
+type SelectivelyVisibleDisasterPeriod struct {
```
Comment (lhl2617) [inline internal/common/config/helpertypes.go]: these should be `SelectivelyVisibleUint`

File: internal/common/config/helpertypes.go [hunk 3]
```diff
@@ -25,6 +25,16 @@ type SelectivelyVisibleResources struct {
...
+	Period uint
+	Valid  bool
+}
+
+type SelectivelyVisibleStochasticDisaster struct {
```
Comment (lhl2617) [inline internal/common/config/helpertypes.go]: `SelectivelyVisibleBool`, similarly.

File: params.go [hunk 4]
```diff
@@ -169,6 +169,16 @@ var (
...
+	disasterPeriodVisible = flag.Bool(
+		"disasterPeriodVisible",
+		true,
+		"Whether disasterPeriod is visible to agents",
+	)
+	disasterStochasticPeriodVisible = flag.Bool(
+		"disasterStochasticPeriodVisible",
+		true,
+		"Whether stochasticPeriod is visible to agents",
+	)
```
Comment (lhl2617) [inline params.go]: I'd move this below the variable where it makes (in)visible. 
Also @JamesTev disasterPeriodStochastic or disasterStochasticPeriod?

Comment (JamesTev) [reply params.go]: Not too fussed. `disasterStochasticPeriod` is how it was defined originally IIRC but the former kind of makes more sense. Up to Eirik.

**Pull request merged**